### PR TITLE
Kce 43 update datamux

### PR DESCRIPTION
**Pull request by mharshe**
Updates for event emitter and datamultiplexer. Also updates spdlog to 1.8.5.

This PR is big but the idea is to bring it up to date with kpe-core (but without the propriety stuff).

### CODE REVIEW

**Review by pablogh-2000 — COMMENTED** [review]
It will be hard to review. I think it we pass the CI, including the dependant repos, we can approve

**Review by pablogh-2000 — APPROVED** [review]

File: core/modules/core/include/klepsydra/core/service_stats.h [hunk 1]
```diff
@@ -45,19 +45,25 @@ struct ServiceStats : public FunctionStats
         : FunctionStats(serviceName)
         , _totalRunningTimeMs(0)
         , _running(false)
-        , _started(false)
     {}
 
     /*!
-     * @brief startTimeWatch
+     * \brief start
```
Comment (paulagb8894) [inline core/modules/core/include/klepsydra/core/service_stats.h:51]: Should be @brief start instead \brief start ?

File: core/modules/core/include/klepsydra/core/subscription_stats.h [hunk 2]
```diff
@@ -39,18 +39,18 @@ struct SubscriptionStats : public FunctionStats
 {
     /*!
      * @brief _subscriberName
-     * @param listenerName name of the listener
-     * @param subscriberName name of the subscriber containing the listener
+     * @param subscriberName name name of the listener
```
Comment (paulagb8894) [inline core/modules/core/include/klepsydra/core/subscription_stats.h:42]: name of the subscriber containing the  listener instead name of the listener

File: core/modules/core/include/klepsydra/core/subscription_stats.h [hunk 3]
```diff
@@ -39,18 +39,18 @@ struct SubscriptionStats : public FunctionStats
...
+     * @param listenerName of the subscriber containing the listener
```
Comment (paulagb8894) [inline core/modules/core/include/klepsydra/core/subscription_stats.h:43]: name of the listener instead name of the subscriber containing the  listener

File: core/modules/core/include/klepsydra/core/unsafe_multi_listener_event_emitter.h [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+                    found = true;
+                    break;
+                }
+            }
+            if (found) {
+                _listenersNameMap.erase(listenerName);
+                if (container && (_listenerStats.find(listenerId) != _listenerStats.end())) {
+                    container->detach(_listenerStats[listenerId].get());
+                    _listenerStats.erase(listenerId);
+                }
```
Comment (paulagb8894) [inline core/modules/core/include/klepsydra/core/unsafe_multi_listener_event_emitter.h:141]: I think we should add the subscriberNameMap removal as in the PR https://github.com/klepsydra-technologies/kpe-core/pull/48/files

Comment (mharshe) [reply core/modules/core/include/klepsydra/core/unsafe_multi_listener_event_emitter.h:141]: Updated to include this PR

**Pull request merged**